### ISSUE TITLE
lintr stopped showing any markup in ESS

### ISSUE THREAD

**Issue by prosoitos** [flycheck/flycheck#1442]
`lintr` was working without any problem in ESS.

I installed the development branch a few weeks ago and I had no more markup in ESS (problems would not get detected). No error message. And `flycheck-verify-setup` looking normal. Showing the error list shows no error (while there are some in the buffer) and the errors don't get highlighted. I opened issue jimhester/lintr#308 then.

I reverted to the CRAN version of `lintr` and the problem went away.

Now, today, while there was no new release of `lintr`, I am having the same issue (now the errors do not get detected, whether I am using the stable or the development branch of `lintr`). So I suspect that it has nothing to do with `lintr` and is due to something in emacs.

Info about my system:

GNU Emacs 25.3.1 (x86_64-pc-linux-gnu, GTK+ Version 3.22.26) of 2018-02-08

Flycheck version: 32snapshot (package: 20180326.836)

OS: 4.15.13-1-ARCH

Customisations: Besides a few face customization, I have `(setq flycheck-global-modes '(not emacs-lisp-mode))`. I changed this today, but it shouldn't affect what is happening in ESS.

Output of `flycheck-verify-setup`:
```Syntax checkers for buffer test_file.R in ess-mode:

  r-lintr
    - may enable:    yes
    - predicate:     t
    - executable:    Found at /usr/bin/R
    - lintr library: present

Flycheck Mode is enabled. Use C-u ; 0 x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20180326.836)
Emacs version:    25.3.1
System:           x86_64-pc-linux-gnu
Window system:    x
```
Thank you!

**Comment by cpitclaudel** [comment]
Thanks for the report. Can you try running C-c ! C-c for that checker? It'd help to see what output it produces.

**Comment by prosoitos** [comment]
Oh, I didn't even know about this function. I seem to have some path problem maybe?

```
-*- mode: compilation; default-directory: "/home/data/geek/r/" -*-
Compilation started at Fri Mar 30 21:03:12

R --slave --restore --no-save -e library\(lintr\)\;try\(lint\(commandArgs\(TRUE\)\,\ cache\=TRUE\,\ with_defaults\(commented_code_linter\ \=\ NULL\,\ spaces_left_parentheses_linter\ \=\ NULL\,\ trailing_whitespace_linter\ \=\ NULL\,\ trailing_blank_lines_linter\ \=\ NULL\)\)\) --args /home/data/geek/r/test_file.R
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/usr/lib/R/library/stringi/libs/stringi.so':
  libicui18n.so.60: cannot open shared object file: No such file or directory

Compilation finished at Fri Mar 30 21:03:12
```

**Comment by cpitclaudel** [comment]
Yes, indeed. Does the same command run OK in a terminal?

**Comment by prosoitos** [comment]
(I forgot to mention, but I have the same issue when running `emacs -Q`)

**Comment by prosoitos** [comment]
> Does the same command run OK in a terminal?

Which command?

**Comment by cpitclaudel** [comment]
> Which command?

Sorry, this one:

```
R --slave --restore --no-save -e library\(lintr\)\;try\(lint\(commandArgs\(TRUE\)\,\ cache\=TRUE\,\ with_defaults\(commented_code_linter\ \=\ NULL\,\ spaces_left_parentheses_linter\ \=\ NULL\,\ trailing_whitespace_linter\ \=\ NULL\,\ trailing_blank_lines_linter\ \=\ NULL\)\)\) --args /home/data/geek/r/test_file.R
```

**Comment by prosoitos** [comment]
Oh, I see.

No. Same problem:

```
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/usr/lib/R/library/stringi/libs/stringi.so':
  libicui18n.so.60: cannot open shared object file: No such file or directory

```

**Comment by cpitclaudel** [comment]
OK. I've never run into this problem (it seems to be independent from Emacs), but it looks like it's not uncommon on Arch Linux; is that what you're running? If so, see https://unix.stackexchange.com/questions/407473/error-while-loading-shared-libraries-libicuuc-so-59-cannot-open-shared-object or https://www.reddit.com/r/archlinux/comments/69x9ah/inox_stopped_working_few_days_ago_libicui18nso58/

**Comment by prosoitos** [comment]
Thank you very much for your time looking into this! Yes, I am running Arch. I'll explore those links and try to find a solution.

**Comment by prosoitos** [comment]
Update:

Rebuilding `stringi` solved the problem and now `lintr` from CRAN is working fine.

Still not working with the development version though, but I now get this error message:

```
Error in gzfile(file, "wb") : cannot open the connection
In addition: Warning messages:
1: In dir.create(path) :
  cannot create dir '/home/marie/.R/lintr_cache', reason 'No such file or directory'
2: In gzfile(file, "wb") :
  cannot open compressed file '/home/marie/.R/lintr_cache/d24099e85bc2a18cc26157c7be9dbabbd1282b86', probable reason 'No such file or directory'
```
I guess this probably has to do with `lintr` and not `flycheck`, so I updated the issue jimhester/lintr#308. If this has nothing to do with `flycheck`, I guess you can close this one.

In any case, thank you very much for your comments!!! They have helped me get started in exploring the problem and I can at least run the stable version again now :slightly_smiling_face:

**Comment by prosoitos** [comment]
Even if this is not `flycheck` related, I thought that I'd update with the solution to my `lintr` problem in case someone ever ends here with a similar issue:

Turned out, I only had to create the `/home/user/.R/lintr_cache` to have it work... Not sure why this is necessary on the development version, but not on the CRAN version.

Anyhow, thank you for `flycheck` and for maintaining it! :slightly_smiling_face:

**Comment by colman-humphrey** [comment]
note for people who are dummies like me: @prosoitos solution is great, and she means to create the `.R/lintr_cache` directory, not as a file... Also, probably no surprise, but on macOS the usual home (`~/`, or `/Users/<user/`) works
